ignite/metrics/epoch_metric.py

```diff
@@ -113,6 +116,12 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
     def compute(self) -> None:
         _prediction_tensor = torch.cat(self._predictions, dim=0)
         _target_tensor = torch.cat(self._targets, dim=0)
+
+        if not self._is_reduced:
+            _prediction_tensor = idist.all_gather(_prediction_tensor)
+            _target_tensor = idist.all_gather(_target_tensor)
+            self._is_reduced = True
+
         return self.compute_fn(_prediction_tensor, _target_tensor)
```

Comment:
To reduce computations on CPU we can apply compute_fn only on a single rank and broadcast the result to all processes.